spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/pulsar/PulsarPropertiesMapper.java

```diff
@@ -137,14 +137,25 @@ private String getAuthenticationParamsJson(Map<String, String> params) {
 		try {
 			return sortedParams.entrySet()
 				.stream()
-				.map((entry) -> "\"%s\":\"%s\"".formatted(entry.getKey(), entry.getValue()))
+				.map((entry) -> "\"%s\":\"%s\"".formatted(entry.getKey(), escapeJson(entry.getValue())))
 				.collect(Collectors.joining(",", "{", "}"));
 		}
 		catch (Exception ex) {
 			throw new IllegalStateException("Could not convert auth parameters to encoded string", ex);
 		}
 	}
 
+	private String escapeJson(String raw) {
```

Comment:
Because we do not directly bring in a JSON library (e.g. Gson or Jackson) we can not use the simple facilities that these libs have to "encode a JSON string". 

Here were the options:

## Use JSON lib
Add required dependency on Jackson or Gson and use their built-in methods

- ✅ solid encode impl maintained by JSON lib
- ❌ ❌ ❌ have to bring in direct dep on JSON lib and this forces it upon our users (not an option)
 
## Add feature to Spring Boot JsonParser
Add "encode a JSON string" to our `JsonParser` abstraction

- ✅ can be used in other areas of Spring Boot codebase
- ❌ created and maintained by us (more work)
- ❌ impl no more solid than if it were a private impl inside the mapper
- ❌ nowhere else in the Boot codebase needs this AFAIK
 
## Hand-roll basic encoder
Inline a simple encode JSON facility in the mapper.

- ❌ basic impl maintained by us
- ❌ ✅ not a perfect solution but will solve the majority of the cases
- ✅ ✅ no direct dep on JSON lib
- ✅ private impl localized to the mapper